GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -371,6 +371,7 @@ public bool RowIsRelative(int rowIndex)
         [CanBeNull]
         public GitRevision GetRevision(int rowIndex)
         {
+            _revisionGraph.CacheTo(rowIndex, /*no update of _orderedRowCache necessary*/ -1)
```

Comment:
@spdr870, should we update the cache in chunks? Of which size?

Reply:
No, the cache you update here is only an ordered list of all the revisions. This is really fast to build. Even 200.000 commits are ordered in ~150ms. The second cache is more expensive, because the rows are being build there.

Reply:
Good.

My idea was that the Cache could cache some more rows when the requested row has not been cached yet. Then the Cache will not need to update everytime the next row is requested.